### PR TITLE
add http caching and warming for forms controller

### PR DESCRIPTION
**Pull request by buddhaCode**
### 1. Why is this change necessary?

Contact forms are not taken into account for HTTP caching, warming and invalidation.

### 2. What does this change do, exactly?

This change adds the possibility to warm up forms for HTTP cache. Changed forms and form fields will invoke the cache invalidation for the respecting form entry. Form route will be cached by default.

### 3. Describe each step to reproduce the issue or behaviour.

1. Warm up HTTP cache for forms
2. Enabled http cache debug option in config.php
```
    'httpcache' => [
        'debug' => true,
    ],
```
3. Open desired contact form in storefront
4. Check HTTP response headers and find, that request is served from cache (fresh)
5. Change form name and/or add a new field
6. Open desired contact form in storefront, again (reload and flush browser cache, to be on the safe side)
7. Changes are reflected in what you see
8. Check HTTP response headers and find, that request isn't served from cache (miss, store)

### 4. Please link to the relevant issues (if any).

Plugin with a backport is available under https://github.com/buddhaCode/BucuFormsHttpCache

### 5. Which documentation changes (if any) need to be made because of this PR?

Update screenshot in chapter "Shopware 5 - User > Konfiguration > Einstellungen > Cache / Performance-Modul > Einstellungen > HTTP-Cache > Cache Warmer"

https://docs.shopware.com/de/shopware-5-de/einstellungen/cache-performance-modul#http-cache

### 6. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: engine/Shopware/Plugins/Default/Core/HttpCache/Bootstrap.php [hunk 1]
```diff
@@ -698,6 +707,12 @@ public function onPostPersist(Enlight_Event_EventArgs $eventArgs)
             case Shopware\Models\Site\Site::class:
                 $cacheIds[] = 's' . $entity->getId();
                 break;
+            case Shopware\Models\Form\Form::class:
+                $cacheIds[] = 'f' . $entity->getId();
+                break;
+            case Shopware\Models\Form\Field::class:
+                $cacheIds[] = 'f' . $entity->getFormId();
```
Comment (shyim) [inline engine/Shopware/Plugins/Default/Core/HttpCache/Bootstrap.php]: Why do they have the same cacheId?

**Comment by shyim** [comment]
Can you change the target to 5.6?
Adding http cache is a bc. Some plugins could extend the forms and expect they are not cached.

File: _sql/migrations/1463-add-http-cache-events-and-routes-for-forms.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * The licensing of the program under the AGPLv3 does not imply a
+ * trademark license. Therefore any rights, title and interest in
+ * our trademarks remain entirely with us.
+ */
+
+class Migrations_Migration1463 extends Shopware\Components\Migrations\AbstractMigration
+{
+    public function up($modus)
+    {
+        $values = $this->connection->query("SELECT v.id, v.value FROM s_core_config_values v INNER JOIN s_core_config_elements e ON e.id = v.element_id WHERE e.name = 'cacheControllers'")
```
Comment (shyim) [inline _sql/migrations/1463-add-http-cache-events-and-routes-for-forms.php]: Single Quotes :)

Comment (buddhaCode) [reply _sql/migrations/1463-add-http-cache-events-and-routes-for-forms.php]: Done

**Comment by buddhaCode** [comment]
I rebased to 5.6 and changed the target branch for the PR.

**Comment by buddhaCode** [comment]
There is a small issue with the renaming of the forms controller viewport from "ticket" to "forms". I'm not sure how to handle this and make it work for existing installations. Please have a look at the [paramter](https://github.com/buddhaCode/BucuFormsHttpCache#technical-information) and my [comment here](https://github.com/shopware/shopware/commit/904f74f29260169b9886814cf825d2c0384f8d37#r33178589)

**Comment by shyim** [comment]
Your rebase seems broken

**Pull request closed without merging**